internal/services/zero_trust_access_service_token/resource.go

```diff
@@ -151,13 +155,20 @@ func (r *ZeroTrustAccessServiceTokenResource) Update(ctx context.Context, req re
 		return
 	}
 	data = &env.Result
+	env.Result.ClientSecret = secret
+	data.ClientSecret = secret
+	resp.State.SetAttribute(ctx, path.Root("client_secret"), &secret)
+	resp.Diagnostics.Append(diags...)
```

Comment:
i don't think this is doing anything as the `data.ClientSecret` assignment above is passed into the `resp.State.Set` below doing this exact thing again.